src/Microsoft.OData.Core/ODataStringEscapeOption.cs

```diff
@@ -0,0 +1,25 @@
+//---------------------------------------------------------------------
+// <copyright file="ODataStringEscapeOption.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData
+{
+    /// <summary>
+    /// Enum type to specify how to escape string value when writing JSON text.
+    /// </summary>
+    public enum ODataStringEscapeOption
+    {
+        /// <summary>
+        /// All non-ASCII and control characters (e.g. newline) are escaped.
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1704", Justification = "Ascii is meaningful")]

```

Comment:
-> "EscapeNonAsciiAndControls", because description above indicates that non-ASCII doesn't contains control chars.

Reply:
#Won't fix.

Reply:
why won't fix? I believe "EscapeNonAsciiAndControls" is more aligned with the description above.

Reply:
Use the same enum member in JSON.NET.

Reply:
well, for people not referencing JSON.NET, the discrepancy between this enum value and it description is confusing.

Reply:
Where's the confusing? See the same comments: https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/StringEscapeHandling.cs#L38-L41

Reply:
I am fine with the current name; control characters are always escaped; there is no way to change that.  so the flag only affects ascii characters.  Since there is already a precedent with NewtonSoft might as well follow that.

---
In reply to: [214485980](https://github.com/OData/odata.net/pull/1242#discussion_r214485980) [](ancestors = 214485980)

Reply:
I'm fine with the current name.

---
In reply to: [215054974](https://github.com/OData/odata.net/pull/1242#discussion_r215054974) [](ancestors = 215054974)

Reply:
In our changes, {EscapeNonAscii, EscapeOnlyControls} are used.
In JSON.NET, {Default, EscapeNonAscii, ...} are used, without specifying explicitly 'Controls'.
That is why I think EscapeNonAsciiAndControls is more clear for our changes, also aligned with description.
This Enum is part of public API, where we should be as clear as possible.